python/cudf/cudf/core/series.py

```diff
@@ -284,6 +284,15 @@ def __setitem__(self, key, value):
             to_dtype = np.result_type(value.dtype, self._frame._column.dtype)
             value = value.astype(to_dtype)
             if to_dtype != self._frame._column.dtype:
+                # Do not remove until pandas-3.0 support is added.
+                warnings.warn(
```

Comment:
Would cudf allow

```
ser = pd.Series([1, 2])
ser[0] = np.nan
```
?

In pandas, this is still allowed to cast int to float since nan is a missing value

Reply:
This isn't possible in cudf, we don't allow setting `np.nan` to a non-floating column.